### PR TITLE
Linux network plugin: Handle IPv6 next hops for IPv4 routes

### PR DESCRIPTION
**Pull request by cooperlees**
## Description
- Add handling for `via` where the inet6 address family is specified
- Add rspec test to show working

I suspect this only fixes part of the issue for #1474 

## Related Issue
Issue: https://github.com/chef/ohai/issues/1474

## Types of changes
- [x ] Bug fix (non-breaking change which fixes an issue)

Signed-off-by: Cooper Lees <me@cooperlees.com>

### CODE REVIEW

File: lib/ohai/plugins/linux/network.rb [hunk 1]
```diff
@@ -105,6 +105,10 @@ def check_routing_table(family, iface, default_route_table)
         route_entry = Mash.new(destination: route_dest,
                                family: family[:name])
         %w{via scope metric proto src}.each do |k|
+          if k == "via" && route_ending =~ /\bvia\sinet6\s+([^\s]+)/
+            route_entry[k] = $1
+            next
+          end
           # http://rubular.com/r/pwTNp65VFf
           route_entry[k] = $1 if route_ending =~ /\b#{k}\s+([^\s]+)/
         end
```
Comment (phiggins) [inline lib/ohai/plugins/linux/network.rb:114]: There doesn't seem to be a reason this new code needs to be inside the loop.

```suggestion
        %w{via scope metric proto src}.each do |k|
          # http://rubular.com/r/pwTNp65VFf
          route_entry[k] = $1 if route_ending =~ /\b#{k}\s+([^\s]+)/
        end
        route_entry["via"] = $1 if route_ending =~ /\bvia\sinet6\s+([^\s]+)/          
```

Comment (phiggins) [reply lib/ohai/plugins/linux/network.rb:114]: The other regex uses `\s+` for whitespace so the new regex probably should as well.

File: lib/ohai/plugins/linux/network.rb [hunk 2]
```diff
@@ -105,6 +105,10 @@ def check_routing_table(family, iface, default_route_table)
         route_entry = Mash.new(destination: route_dest,
                                family: family[:name])
         %w{via scope metric proto src}.each do |k|
+          if k == "via" && route_ending =~ /\bvia\sinet6\s+([^\s]+)/
+            route_entry[k] = $1
+            next
```
Comment (tas50) [inline lib/ohai/plugins/linux/network.rb:110]: Couldn't this just be an if / else and avoid the next?

File: lib/ohai/plugins/linux/network.rb [hunk 3]
```diff
@@ -108,6 +108,7 @@ def check_routing_table(family, iface, default_route_table)
           # http://rubular.com/r/pwTNp65VFf
           route_entry[k] = $1 if route_ending =~ /\b#{k}\s+([^\s]+)/
         end
+        route_entry["via"] = $1 if route_ending =~ /\bvia\s+inet6\s+([^\s]+)/
```
Comment (tas50) [inline lib/ohai/plugins/linux/network.rb:111]: can you add a rubular permalink like the one above for this regex so we can see exactly how it's matching?

Comment (cooperlees) [reply lib/ohai/plugins/linux/network.rb:111]: Don't the unittests show that? And even show them much clearer. I feel that's way more useful than a URL comment ... But I can add, will try work it out.

**Comment by tas50** [comment]
Thanks for the updates @cooperlees

**Pull request merged**